script/vsts/platforms/templates/preparation.yml

```diff
@@ -64,4 +64,4 @@ steps:
       cd script\vsts
       npm install
     displayName: Install Windows build dependencies
-    condition: eq(variables['Agent.OS'], 'Windows_NT')
+    condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['buildArch'], 'x86'))
```

Comment:
This is incorrect.

Reply:
I would like us to consolidate these variables.

`buildArch` is set and then `BUILD_ARCH` is set equal to `buildArch`. As far as I can tell, this is unnecessary; Only variables set from Azure DevOps UI need to be UPPERCASE_WITH_UNDERSCORES. Since these are set in yaml, they may be camelCase.

As far as I can tell:

- `buildArch` and `BUILD_ARCH` are only set on Windows (x86 and x64)
- `buildArch` and `BUILD_ARCH` are _always_ available on Windows, whether as Azure DevOps variables, or as environment variables
- `buildArch` and `BUILD_ARCH` hold the same value at any given moment

Therefore, I'd like us to pick one and settle on it so we can be consistent and make the CI less confusing. No need to have two different variables for the exact same value if one is good enough.

One thing to note: `script/vsts/windows-run.js` looks for `BUILD_ARCH` (capital case with underscores), so we need to standardize on that, OR edit the script to look for `process.env.buildArch` instead of `process.env.BUILD_ARCH`.

It would be a big priority for me to consolidate these due to the unnecessary complexity and confusion.

Reply:
And in case that's not what you mean, how is it incorrect?

One of my goals in this PR was to make sure the `script/vsts/node_modules` dependencies, and `windows-run.js`, are totally not needed for Windows x64.

If I recall correctly, the only thing still not working is the "Upload crash reports" under Windos x64, which I suppose is still a blocker, because I don't think upstream will merge if that's broken.

If need be, I can give up on eliminating `windows-run.js` on Windows x64. It did turn out to be more complicated than expected.

In that case, we should still switch `cd script/vsts && npm install` to `cd script/vsts && npm ci` to be slightly faster. (We don't cache those dependencies, so there is no downside to `ci` vs `install`).

Reply:
If this step is only needed for x86 that makes sense. You should rename the display name and comment as well then to be specific to `x86`